knowledge graph/kg_agent.py

```
@@ -0,0 +1,351 @@
+        self.driver = neo4j.AsyncGraphDatabase.driver(
+            uri, auth=(user, password)
+        )
+
+    def _validate_label(label: str) -> None:
+        # Only allow ASCII letters, digits and underscores, and must start with a letter
+        if not re.match(r'^[A-Za-z][A-Za-z0-9_]*$', label):
+            raise ValueError(f"Invalid entity type: {label}")
+    
+    async def create_entity(self, entity_type: str, properties: Dict[str, Any]):
+        """Create a new entity node in the knowledge graph"""
+        _validate_label(entity_type)
```

Ruff (0.12.2)

Undefined name `_validate_label`

(F821)